com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StopAppStep.java

```diff
@@ -58,12 +59,28 @@ protected String getStepErrorMessage(ProcessContext context) {
 
     @Override
     public List<HookPhase> getHookPhasesBeforeStep(ProcessContext context) {
-        return hooksPhaseBuilder.buildHookPhases(Collections.singletonList(HookPhase.BEFORE_STOP), context);
+        List<HookPhase> hookPhases = new ArrayList<>();
+        hookPhases.add(HookPhase.BEFORE_STOP);
+        ProcessType processType = processTypeParser.getProcessType(context.getExecution());
+        if (ProcessType.BLUE_GREEN_DEPLOY.equals(processType)) {
+            hookPhases.add(HookPhase.APPLICATION_BEFORE_STOP_IDLE);
+        } else {
+            hookPhases.add(HookPhase.APPLICATION_BEFORE_STOP_LIVE);
+        }
+        return hooksPhaseBuilder.buildHookPhases(hookPhases, context);

```

Comment:
Try extracting the whole block of code into another method and try to re-use it below, in the method getHookPhasesAfterStep, because it seems like a copy-pasted below?